src/python/pants/backend/cc/subsystems/compiler.py

```diff
@@ -0,0 +1,135 @@
+            These flags are merged with the target-level defines, with the target-level flags taking precedence.
+            """
+        ),
+    )
+
+    c_defines = StrListOption(
+        default=None,
+        help=softwrap(
+            """
+            A list of strings to define in the C preprocessor. Will be prefixed by -D at the command line.
+            These defines are merged with the target-level defines, with the target-level definitions taking precedence.
+            """
+        ),
+    )
+
+    cpp_compile_options = StrListOption(
+        default=["-std=c++11"],
+        help=softwrap(
+            """
+            Flags passed to the C++ compiler.
+            These flags are merged with the target-level defines, with the target-level flags taking precedence.
+            """
+        ),
+    )
+
+    cpp_defines = StrListOption(
+        default=None,
+        help=softwrap(
+            """
+            A list of strings to define in the C++ preprocessor. Will be prefixed by -D at the command line.
+            These defines are merged with the target-level defines, with the target-level definitions taking precedence.
+            """
+        ),
+    )
+
+
+class ExternalCCSubsystem(TemplatedExternalTool, CCOptions):
+    options_scope = "cc-external"
+    name = "cc-external"
+    help = """Options for downloaded CC toolchain support."""
+
+    c_executable = StrOption(
+        default="gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gcc",
+        help=softwrap(
+            """
+            The relative path to the C compiler binary from the downloaded source.
+            E.g. For the ARM gcc-rm toolchain, this value would be: `gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gcc`
+            """
+        ),
+    )
+
+    cpp_executable = StrOption(
+        default="gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-g++",
+        help=softwrap(
+            """
+            The relative path to the C++ compiler binary from the downloaded source.
+            E.g. For the ARM gcc-rm toolchain, this value would be: `gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-g++`
+            """
+        ),
+    )
+
+    default_version = "10.3-2021.10"
+    # NOTE: This is only fully populated for testing - will be emptied before PR
+    default_url_template = ""  # "https://developer.arm.com/-/media/Files/downloads/gnu-rm/{version}/gcc-arm-none-eabi-{version}-{platform}.tar.bz2"
+    default_url_platform_mapping = {
+        # "macos_arm64": "darwin_arm64",
+        "macos_x86_64": "mac",
+        # "linux_x86_64": "linux_amd64",
+    }
+    default_known_versions = [
+        "10.3-2021.10|macos_x86_64|fb613dacb25149f140f73fe9ff6c380bb43328e6bf813473986e9127e2bc283b|158961466",
+    ]
+
+
+class CCSubsystem(Subsystem, CCOptions):
+    options_scope = "cc"
+    name = "cc"
+    help = """Options for system CC toolchain support."""
+
+    c_executable = StrListOption(
+        default=["clang", "gcc"],
```

Comment:
These defaults assume the user's system has their gcc/clang versions aliased to these names - which I think is a fair assumption. They can always be specified as needed to something like `clang-15`